Provide Microsoft.Extentions.FileProviders.Abstractions package readme (#93489)
dotnet/runtime: src/libraries/Microsoft.Extensions.FileProviders.Abstractions/src/PACKAGE.md

<!-- A description of the package and where one can find more documentation -->

> `Microsoft.Extensions.FileProviders.Abstractions` serves as the foundation for creating file providers in .NET, offering core abstractions to develop custom file providers.
> It enables fetching files from various sources.

Reviewer: ```suggestion
Serves as the foundation for creating file providers in .NET, offering core abstractions to develop custom file providers capable of fetching files from various sources.
```

## Key Features

<!-- The key features of this package -->

* Core abstractions for creating and managing file providers.
* Flexibility to develop custom file providers for fetching files from distinct sources.

## How to Use

<!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->

The abstractions provided by this package can be utilized to create custom file providers. Here's a simplified example of implementing a custom file provider using the `IFileProvider` interface:

```csharp
using Microsoft.Extensions.FileProviders;
using System;

public class CustomFileProvider : IFileProvider
{
    // Implement methods defined in IFileProvider
}

class Program
{
    static void Main(string[] args)
    {
        IFileProvider fileProvider = new CustomFileProvider();
        // Use the custom file provider
    }
}

> ```

Reviewer: Can we have a functional IFileProvider instead? e.g: https://github.com/dazinator/Dazinator.Extensions.FileProviders/blob/7c00e96b0b4a7e40342ea2abc1cf1d418be2b660/src/Dazinator.Extensions.FileProviders/InMemory/InMemoryFileProvider.cs

Reviewer: Yes, I agree that's a large example for this document. Nvm my suggestion.

## Main Types

<!-- The main types provided in this library -->

The main types provided by this library are:

* `IFileProvider`
* `IDirectoryContents`
* `IFileInfo`

> * `NullFileProvider`

Reviewer: Use fully qualified names.

## Additional Documentation

<!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->

> * [Conceptual documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/change-tokens)
> * [API documentation](https://learn.microsoft.com/nl-nl/dotnet/api/microsoft.extensions.fileproviders)

Reviewer: ```suggestion
* [Conceptual documentation](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/file-providers)
* [Detect changes with change tokens](https://learn.microsoft.com/aspnet/core/fundamentals/change-tokens)
* [API documentation](https://learn.microsoft.com/dotnet/api/microsoft.extensions.fileproviders)
```

<!-- A description of the package and where one can find more documentation -->

Serves as the foundation for creating file providers in .NET, offering core abstractions to develop custom file providers capable of fetching files from various sources.

## Key Features

<!-- The key features of this package -->

* Core abstractions for creating and managing file providers.
* Flexibility to develop custom file providers for fetching files from distinct sources.

## How to Use

<!-- A compelling example on how to use this package with code, as well as any specific guidelines for when to use the package -->

The abstractions provided by this package can be used to create custom file providers.

Here's a simplified example of implementing a custom file provider using the `IFileProvider` interface:

```csharp
using Microsoft.Extensions.FileProviders;
using System;

public class CustomFileProvider : IFileProvider
{
    // Implement methods defined in IFileProvider
}

class Program
{
    static void Main(string[] args)
    {
        IFileProvider fileProvider = new CustomFileProvider();
        // Use the custom file provider
    }
}

> ```

Reviewer: How about we change this section to something similar to [`MS.Extensions.DependencyInjection.Abstractions`](https://github.com/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/PACKAGE.md#how-to-use)?

```md
This package is typically used with an implementation of file provider abstractions, such as `Microsoft.Extensions.FileProviders.Composite` and `Microsoft.Extensions.FileProviders.Physical`.
```
